### PR TITLE
Include support for ecto_sql & phoenix_ecto

### PR DESCRIPTION
**Pull request by nurugger07**
* Adds ecto_sql, phoenix_ecto, & postgrex dependencies
* Setups up configs & Repo
* Adds uuid-ossp extension to postgres with migration
* Updates README for database setup steps

Requirements: Have a local copy of postgres running

QA INFO: 
Download and run `mix do ecto.create, ecto.migrate`
Verify the database has been created locally
Verify the extension has been added be running the following from 

```
$ pgsql erlef_website_dev
erlef_website_dev=# select uuid_generate_v4();
           uuid_generate_v4
--------------------------------------
 38852d8a-3bbe-40c1-865e-7fa6a6065418
```

The result will be different but look should be the same :)

### CODE REVIEW

File: test/support/data_case.ex [hunk 1]
```diff
@@ -0,0 +1,19 @@
+defmodule Data.DataCase do
```
Comment (starbelly) [inline test/support/data_case.ex]: This is a nit, but I'd say let's call this Erlef.DataCase

File: config/test.exs [hunk 2]
```diff
@@ -12,3 +12,9 @@ config :erlef, Erlef.Mailer, adapter: Swoosh.Adapters.Test
 
 # Print only warnings and errors during test
 config :logger, level: :warn
+
+config :erlef, Erlef.Data.Repo,
+  database: "erlef_website_test",
+  pool: Ecto.Adapters.SQL.Sandbox,
+  migration_primary_key: [id: :uuid, type: :binary_id],
```
Comment (starbelly) [inline config/test.exs]: Are we sure we want UUID for primary keys? I'm not sure the database will ever be big enough or we will have a situation where we can't reliable increment an integer PK... but I'm all ears for the argument.

Comment (nurugger07) [reply config/test.exs]: i usually opt for UUID from the start. I've been bitten enough over the years using integers for primary keys. The uniqueness factor plays a big part in that. Plus, the effort is minimal to implement from the start and far more difficult to implement later.

Comment (starbelly) [reply config/test.exs]: 👍

**Pull request merged**